lib/model/recent_senders.dart

```diff
@@ -68,21 +68,61 @@ class RecentSenders {
       [senderId] ??= MessageIdTracker()).add(messageId);
   }
 
+  /// Handles channel/topic updates when messages are moved.
+  ///
+  /// This is a no-op if no message move happened.
+  void handleUpdateMessageEvent(UpdateMessageEvent event, Map<int, Message> cachedMessages) {
+    if (event.moveData == null) {
+      return;
+    }
+    final UpdateMessageMoveData(
+      :origStreamId, :newStreamId, :origTopic, :newTopic) = event.moveData!;
+
+    final messagesBySender = _groupStreamMessageIdsBySender(event.messageIds, cachedMessages);
+    final sendersInChannel = streamSenders[origStreamId];
```

Comment:
```suggestion
    final sendersInStream = streamSenders[origStreamId];
```

Given all the other "stream" names here, plus the otherwise very similar `handleDeleteMessageEvent` below which it's useful to compare this code to, I think it's clearest if these particular locals stick with the "stream" names for now. We'll switch them all to "channel" names as part of the post-launch sweep we have in the tracker.